source/extensions/upstreams/http/http/BUILD

```diff
@@ -33,6 +33,7 @@ envoy_cc_library(
     hdrs = [
         "upstream_request.h",
     ],
+    visibility = ["//visibility:public"],
```

Comment:
Seems a bit odd for me to see cc_library with public visibility, why is this required?

Reply:
I think it's the instruction `envoy_extension_package()` turns the `upstream_request_lib` invisible to other packages. I can double check if I was doing anything wrong in the past.

Do you agree that upstream_request_lib could be utilized by other packages? I really need this upstream_request so that the upstream request could be wrapped in outer upstream request.

Reply:
This is the error message if the visibility is removed.
```
$ bazel test //test/integration:cluster_upstream_extension_integration_test 
ERROR: /home/lambdai/workspace/envoy/test/integration/upstreams/BUILD:12:19: in cc_library rule //test/integration/upstreams:per_host_upstream_config: target '//source/extensions/upstreams/http/http:upstream_request_lib' is not visible from target '//test/integration/upstreams:per_host_upstream_config'. Check the visibility declaration of the former target if you think the dependency is legitimate
ERROR: Analysis of target '//test/integration:cluster_upstream_extension_integration_test' failed; build aborted: Analysis of target '//test/integration/upstreams:per_host_upstream_config' failed
```